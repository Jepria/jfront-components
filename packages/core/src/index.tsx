export * from '@jfront-ui/app'
export * from '@jfront-ui/datepicker'
export * from '@jfront-ui/form'
//export * from '@jfront-ui/grid'
export * from '@jfront-ui/hooks'
//export * from '@jfront-ui/input'
export * from '@jfront-ui/layout'
//export * from '@jfront-ui/pagingbar'
//export * from '@jfront-ui/tabpanel'
//export * from '@jfront-ui/toolbar'
