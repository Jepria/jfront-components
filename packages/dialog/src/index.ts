export * from './SimpleDialog'
export * from './AlertDialog'
export * from './FormDialog'
export * from './ErrorDialog'
export * from './WarningDialog'