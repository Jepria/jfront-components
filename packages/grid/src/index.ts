export * from './common'
export * from './jepgrid'
