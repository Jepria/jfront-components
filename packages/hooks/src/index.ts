export * from "./useOnClickOutside"
export * from "./useRouteMapping"
export * from "./useTree/"
export * from "./useTree/types"
export * from "./useFilter"
