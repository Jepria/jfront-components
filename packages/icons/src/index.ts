export * from "./Add"
export * from "./Delete"
export * from "./Edit"
export * from "./Exclamation"
export * from "./First"
export * from "./Last"
export * from "./Loading"
export * from "./Next"
export * from "./OpenIcon"
export * from "./Prev"
export * from "./Refresh"
export * from "./Save"
export * from "./Search"
export * from "./SettingsSmall"
export * from "./Split"
export * from "./View"
