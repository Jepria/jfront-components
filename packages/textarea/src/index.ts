export interface AreaProps {
  error?: string
  name?: string
  isLoading?: boolean
  resize?: string
}

export * from "./TextArea"
