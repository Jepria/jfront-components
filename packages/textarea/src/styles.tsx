import styled from "styled-components"

export interface StyledAreaProps {
  error?: boolean
  focused?: boolean
  disabled?: boolean
}

export interface StyledTextAreaProps {
  error?: boolean
  focused?: boolean
  resize?: string
}

export const StyledDiv = styled.div<StyledAreaProps>`
  display: inline-flex;
  white-space: nowrap;
  align-items: center;
  justify-content: center;
  text-align: left;
  min-width: 150px;
  height: ${(props) => props.theme.textArea.height};
  border-radius: ${(props) => props.theme.textArea.borderRadius};
  margin: ${(props) => props.theme.textArea.margin};
  padding: ${(props) => props.theme.textArea.padding};
  font-size: ${(props) => props.theme.fontSize.medium};
  font-family: ${(props) => props.theme.fontFamily};
  background: ${(props) =>
    props.error
      ? props.theme.textArea.errorBgColor
      : props.focused
      ? props.theme.textArea.focusedBgColor
      : props.disabled
      ? props.theme.textArea.disabledBgColor
      : props.theme.textArea.bgColor};
  color: ${(props) =>
    props.error
      ? props.theme.textArea.errorColor
      : props.focused
      ? props.theme.textArea.focusedColor
      : props.disabled
      ? props.theme.textArea.disabledColor
      : props.theme.textArea.color};
  border: ${(props) =>
    `${props.theme.textArea.borderWidth} 
  ${props.theme.textArea.borderStyle} 
  ${
    props.error
      ? props.theme.textArea.errorBorderColor
      : props.focused
      ? props.theme.textArea.focusedBorderColor
      : props.disabled
      ? props.theme.textArea.disabledBorderColor
      : props.theme.textArea.borderColor
  }`};
  ${(props) =>
    props.focused
      ? `box-shadow: 0 0 5px ${
          props.error
            ? props.theme.textArea.errorBorderColor
            : props.theme.textArea.focusedBorderColor
        };`
      : ""}
  &:hover {
    ${(props) =>
      !props.focused && !props.error && !props.disabled
        ? `border: ${props.theme.textArea.borderWidth} 
    ${props.theme.textArea.borderStyle} 
    ${props.theme.textArea.hoverBorderColor};
    color: ${props.theme.textArea.hoverColor};`
        : ""}
  }
`

StyledDiv.defaultProps = {
  theme: {
    fontSize: {
      medium: "12px",
    },
    fontFamily: "tahoma, arial, helvetica, sans-serif",
    textArea: {
      borderWidth: "1px",
      borderStyle: "solid",
      margin: 0,
      padding: 0,
      borderColor: "#ccc",
      borderRadius: 0,
      bgColor: "#fff",
      color: "black",
      focusedBorderColor: "#99bbe8",
      focusedColor: "black",
      focusedBgColor: "#fff",
      errorBorderColor: "red",
      errorColor: "black",
      errorBgColor: "#fff",
      disabledColor: "rgba(0, 0, 0, 0.75)",
      disabledBorderColor: "#ccc",
      disabledBgColor: "rgba(239, 239, 239, 0.5)",
      hoverColor: "black",
      hoverBorderColor: "#99bbe8",
      hoverBgColor: "#fff",
    },
  },
}

export const StyledTextArea = styled.textarea<StyledTextAreaProps>`
  display: inline-flex;
  flex-grow: 1;
  min-width: 0px;
  margin: 0;
  padding: 0;
  padding-left: 3px;
  font-family: tahoma, arial, helvetica, sans-serif;
  font-size: 12px;
  border: 0;
  width: 250px;
  height: 100px;
  box-sizing: border-box;
  &:focus {
    outline: none;
  }
  background: transparent;
  ${(props) =>
    props.resize !== undefined
      ? `resize:` + props.resize + `;`
      : `resize: none`};
`
