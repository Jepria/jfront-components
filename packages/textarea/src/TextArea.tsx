import React, { useState } from "react"
import { AreaProps } from "./index"
import { StyledTextArea, StyledDiv } from "./styles"
import { LoadingImage, ExclamationImage } from "@jfront/ui-icons"

export interface TextAreaProps
  extends React.TextareaHTMLAttributes<HTMLTextAreaElement>,
    AreaProps {}

export const TextArea = React.forwardRef<HTMLDivElement, TextAreaProps>(
  ({ style, className, ...props }, ref) => {
    const [focused, setFocused] = useState(false)

    return (
      <StyledDiv
        className={className}
        focused={focused}
        style={style}
        ref={ref}
        error={props.error !== undefined}
      >
        <StyledTextArea
          {...props}
          onFocus={(e) => {
            if (props.onFocus) {
              props.onFocus(e)
            }
            setFocused(true)
          }}
          onBlur={(e) => {
            if (props.onBlur) {
              props.onBlur(e)
            }
            setFocused(false)
          }}
          /*ref={props.inputRef}
                    type={props.clearButton ? "search" : "text"}*/
        />
        {props.isLoading && <LoadingImage />}
        {props.error !== undefined && <ExclamationImage title={props.error} />}
      </StyledDiv>
    )
  },
)
