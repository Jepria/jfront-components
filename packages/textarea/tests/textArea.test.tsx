import React from "react"
import { render } from "@testing-library/react"
import { TextArea } from "../src"

const vtextValue = "tets"
test("Checking for the existence of an element Tab", () => {
  render(
    <TextArea name="TextAreaName" value={vtextValue}>
      Test
    </TextArea>,
  )
})

test("Matches snapshot ", () => {
  const { asFragment } = render(
    <TextArea name="TextAreaName" value={vtextValue}>
      Test
    </TextArea>,
  )
  expect(asFragment()).toMatchSnapshot()
})
test("isLoading display TextArea ", () => {
  const { asFragment } = render(
    <>
      <TextArea name="TextAreaNameLoading" isLoading value={vtextValue}>
        Текст при загрузке
      </TextArea>
    </>,
  )
  expect(asFragment()).toMatchSnapshot()
})
test("error display TextArea", () => {
  const { asFragment } = render(
    <>
      <TextArea
        name="TextAreaNameError"
        error="Ahtung! Wrong value"
        value={vtextValue}
      >
        Текст при ошибке
      </TextArea>
    </>,
  )
  expect(asFragment()).toMatchSnapshot()
})
test("resize display TextArea ", () => {
  const { asFragment } = render(
    <>
      <TextArea name="TextAreaNameResize" resize="both" value={vtextValue}>
        Изменение размера по горизонтали и вертикали
      </TextArea>
    </>,
  )
  expect(asFragment()).toMatchSnapshot()
})
test("resize horizontal display TextArea", () => {
  const { asFragment } = render(
    <>
      <TextArea
        name="TextAreaNameResize"
        resize="horizontal"
        value={vtextValue}
      >
        Изменение размера по горизонтали
      </TextArea>
    </>,
  )
  expect(asFragment()).toMatchSnapshot()
})
test("resize horizontal display TextArea", () => {
  const { asFragment } = render(
    <>
      <TextArea name="TextAreaNameResize" resize="resize" value={vtextValue}>
        Изменение размера по вертикали
      </TextArea>
    </>,
  )
  expect(asFragment()).toMatchSnapshot()
})
