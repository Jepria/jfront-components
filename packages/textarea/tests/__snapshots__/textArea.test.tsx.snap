// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot  1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm eRgSgo"
  >
    <textarea
      class="sc-AxirZ iexTxl"
      name="TextAreaName"
    >
      tets
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`error display TextArea 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm dXdfdq"
  >
    <textarea
      class="sc-AxirZ iexTxl"
      name="TextAreaNameError"
    >
      tets
    </textarea>
    <img
      class="sc-AxiKw bxCqvx"
      src="data:image/gif;base64,R0lGODlhEAAQAIcAAED/QMJLMsNNNMVPNsNUPcxVPMZXQcddRs5dTNBZQcxiTM5kTtVgR9FjTNFhUNRkUthwWd5wWd1wXd18b919b+FdUuJfU+NhVOViVeZlVudnV+NmW+RnXOdsXehpWOpsWetuWulvX+F0XOVyWep2XexwW+5zXO92XfB4X/F6X+BzYO57Yel9aPN8YPN+YemCau2BaeyBbO6DbuOEdeaIeOqHfeuJfeqNe+yLf+ySf/KAZfSAYvWBYvWCY/SEZ/WEaveGbPaJa/iGbfONdvWOd/eMdfqOd/qUfN6hleWOg+eQhOeRheySiO2SieyYie2flfGTgvCXg/GXhPSZhPCfjvGdkOGkl+Onmu6glfihiPiiifijivOmkfSmkvGlmfGom/Gon/Kpn+uvovOvpvSsofWwo/ezoPezpPC0qPi0oPi2ofi1pfm5pvq8qPu9qfi+sfq/sv3DtfrZ0////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAMAAAAALAAAAAAQABAAAAjBAAEIHEiwoME4R4D8IPLG4EAjMrqkScOFBQyHQqS42RIkiJY2OUYULBLFTY8ec+a4aMHmhoiBcGKY5LEjZQoUJ9ZIECNwCJcsO1amPGGiBBQnEASuUOOjxc2UJUB8CPElgUASZnTgNCFnzgcPGjp4KSDwBZUpRUFUGaMhAwYbSxYIRKPijFQPKTFcsBDGgZWBEWiUaduEiYUKYSgYKMhgBhkcHDbUADMhgMMGD5RgeZIEAQGHAq8oGCDgABLQqAkGBAA7"
      title="Ahtung! Wrong value"
    />
  </div>
</DocumentFragment>
`;

exports[`isLoading display TextArea  1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm eRgSgo"
  >
    <textarea
      class="sc-AxirZ iexTxl"
      name="TextAreaNameLoading"
    >
      tets
    </textarea>
    <img
      class="sc-AxhCb fOTuKT"
      src="data:image/gif;base64,R0lGODlhEAAQALMMAKqooJGOhp2bk7e1rZ2bkre1rJCPhqqon8PBudDOxXd1bISCef///wAAAAAAAAAAACH/C05FVFNDQVBFMi4wAwEAAAAh+QQFAAAMACwAAAAAEAAQAAAET5DJyYyhmAZ7sxQEs1nMsmACGJKmSaVEOLXnK1PuBADepCiMg/DQ+/2GRI8RKOxJfpTCIJNIYArS6aRajWYZCASDa41Ow+Fx2YMWOyfpTAQAIfkEBQAADAAsAAAAABAAEAAABE6QyckEoZgKe7MEQMUxhoEd6FFdQWlOqTq15SlT9VQM3rQsjMKO5/n9hANixgjc9SQ/CgKRUSgw0ynFapVmGYkEg3v1gsPibg8tfk7CnggAIfkEBQAADAAsAAAAABAAEAAABE2QycnOoZjaA/IsRWV1goCBoMiUJTW8A0XMBPZmM4Ug3hQEjN2uZygahDyP0RBMEpmTRCKzWGCkUkq1SsFOFQrG1tr9gsPc3jnco4A9EQAh+QQFAAAMACwAAAAAEAAQAAAETpDJyUqhmFqbJ0LMIA7McWDfF5LmAVApOLUvLFMmlSTdJAiM3a73+wl5HYKSEET2lBSFIhMIYKRSimFriGIZiwWD2/WCw+Jt7xxeU9qZCAAh+QQFAAAMACwAAAAAEAAQAAAETZDJyRCimFqbZ0rVxgwF9n3hSJbeSQ2rCWIkpSjddBzMfee7nQ/XCfJ+OQYAQFksMgQBxumkEKLSCfVpMDCugqyW2w18xZmuwZycdDsRACH5BAUAAAwALAAAAAAQABAAAARNkMnJUqKYWpunUtXGIAj2feFIlt5JrWybkdSydNNQMLaND7pC79YBFnY+HENHMRgyhwPGaQhQotGm00oQMLBSLYPQ9QIASrLAq5x0OxEAIfkEBQAADAAsAAAAABAAEAAABE2QycmUopham+da1cYkCfZ94UiW3kmtbJuRlGF0E4Iwto3rut6tA9wFAjiJjkIgZAYDTLNJgUIpgqyAcTgwCuACJssAdL3gpLmbpLAzEQA7"
    />
  </div>
</DocumentFragment>
`;

exports[`resize display TextArea  1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm eRgSgo"
  >
    <textarea
      class="sc-AxirZ jWPjXJ"
      name="TextAreaNameResize"
    >
      tets
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`resize horizontal display TextArea 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm eRgSgo"
  >
    <textarea
      class="sc-AxirZ bMPzmO"
      name="TextAreaNameResize"
    >
      tets
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`resize horizontal display TextArea 2`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm eRgSgo"
  >
    <textarea
      class="sc-AxirZ bQWygS"
      name="TextAreaNameResize"
    >
      tets
    </textarea>
  </div>
</DocumentFragment>
`;
